contracts/script/FeeContract.s.sol

```diff
@@ -0,0 +1,207 @@
+pragma solidity ^0.8.20;
+
+import { Script } from "forge-std/Script.sol";
+import {
+    ProposeUpgradeResponse,
+    Defender,
+    ApprovalProcessResponse
+} from "openzeppelin-foundry-upgrades/Defender.sol";
+import { Upgrades, Options } from "openzeppelin-foundry-upgrades/Upgrades.sol";
+import { FeeContract as FC } from "../src/FeeContract.sol";
+import { UtilsScript } from "./Utils.s.sol";
+
+/// @notice Deploys an upgradeable Fee Contract using the OpenZeppelin Upgrades plugin.
+contract DeployFeeContractScript is Script {
+    string internal contractName = "FeeContract.sol";
+
+    /// @dev Deploys both the proxy and the implementation contract.
+    /// The proxy admin is set as the owner of the contract upon deployment.
+    /// The `owner` parameter should be the address of the multisig wallet to ensure proper
+    /// ownership management.
+    /// @param owner The address that will be set as the owner of the proxy (typically a multisig
+    /// wallet).
+    function run(address owner)
+        public
+        returns (address payable proxy, address implementationAddress)
+    {
+        string memory seedPhrase = vm.envString("DEPLOYER_MNEMONIC");
+        uint32 seedPhraseOffset = uint32(vm.envUint("DEPLOYER_MNEMONIC_OFFSET"));
+        (address admin,) = deriveRememberKey(seedPhrase, seedPhraseOffset);
+        vm.startBroadcast(admin);
+
+        address proxyAddress =
+            Upgrades.deployUUPSProxy(contractName, abi.encodeCall(FC.initialize, (owner)));
+
+        // Get the implementation address
+        implementationAddress = Upgrades.getImplementationAddress(proxyAddress);
+
+        vm.stopBroadcast();
+
+        return (payable(proxyAddress), implementationAddress);
+    }
+}
+
+/// @notice Upgrades the fee contract first by deploying the new implementation
+/// and then executing the upgrade via the Safe Multisig wallet using the SAFE SDK.
+contract UpgradeFeeContractScript is Script {
+    string internal originalContractName = "FeeContract.sol";
```

Comment:
If we want to use this upgrade more than once, should we also make it so that the `originalContractName` can be derived from an environment variable?

Reply:
thanks @Ayiga , i did that in this update  https://github.com/EspressoSystems/espresso-sequencer/commit/4ac6ca7cfd7492ad82a83177b50cb2a25a00f9fc however, the actual contract still needs to be imported into the script as we see here https://github.com/EspressoSystems/espresso-sequencer/blob/4ac6ca7cfd7492ad82a83177b50cb2a25a00f9fc/contracts/script/FeeContract.s.sol#L67 so there will still need to be changes to the script when the time comes